### ISSUE TITLE
channel::get_user_permissions doesn't give the correct permissions of the bot

### ISSUE THREAD

**Issue by Commandserver** [brainboxdotcc/DPP#400]
**Git commit reference**
6cf7c038d4b37364bde1b654f4fbb65e0a44464f

**Describe the bug**
When calling `channel::get_user_permissions` and passing bot.me, it doesn't give the correct permissions.

**To Reproduce**
```c++
bot.on_message_create([&bot](const dpp::message_create_t &event) {
		auto channel = dpp::find_channel(event.msg.channel_id);
		auto p = channel->get_user_permissions(&event.msg.author);
		cout << "permission: " << p << endl;
		cout << "admin?: " << p.has(dpp::p_administrator) << endl;
	});
```
prints this when i write something in a channel
> permission: 1071698533953
admin?: 0

although my bot has administrator.

**Comment by braindigitalis** [comment]
i cant debug this.
you need to trace it back through the bot's code and figure out where its going wrong, because i dont have your roles, your user setup, etc.
also, bot.me is filled before the cache is filled, try this instead with dpp::find_user() instead of bot.me

**Comment by braindigitalis** [comment]
hmmm
wait, just noticed. 
if you want to know if theyre administrator you dont get the *channel* permissions, you get the base permissions of the guild... administrator is not channel specific.

**Comment by Commandserver** [comment]
> i cant debug this. you need to trace it back through the bot's code and figure out where its going wrong, because i dont have your roles, your user setup, etc. also, bot.me is filled before the cache is filled, try this instead with dpp::find_user() instead of bot.me

hm, doesnt work with dpp::find_user() either

**Comment by braindigitalis** [comment]
probably related to  #403

**Comment by braindigitalis** [comment]
fixed by #403
